TODO.md

```diff
@@ -0,0 +1,23 @@
+## TODO
+
+* remove in-progress state and use row-lock?  this may handle killing upload process more gracefully
+* logging events? upload succeed, upload failed etc.?
+* make sure that a failed/killed process can't result in a blob stuck as "uploading", or if that is possible then there is a way to clean that up from the CLI
+* check what happens with S3 when trying to re-upload identical content - https://groups.google.com/g/jets3t-users/c/i1gusIu5mTw
+* consider NOT deleting blob.content on upload - instead add CLI script to EXPLICITLY delete all uploaded blobs
+* if e.g. 1 blob is missing from s3, and user is trying to export large amount of data, how should we handle?
+  * simpler: just fail completely
+  * more helpful: e.g. inline error message in exported data... but if done badly this could be misleading and the user may never notice that they've got a partial export
+* e2e test is currently not very interesting - could the upload-pending CLI call be async, and we test what happens if we try to fetch attachments while upload is in progress?
+* should config key be default.external.s3blobStore?
+* remove this file
+
```

Comment:
Additional questions:
- Should S3 config be set in central .env and passed through? 

- What should be documented about setting up S3?
  - Minimal recommended setup of S3
  - S3 retention strategy? Stop things from being deleted manually from S3?
  - CLI usage

- Should upload-blobs cli be part automatically part of cron job or should it be fully under admin's control? Manual control (e.g. running upload script or adding to a cron job themselves) can allow uploading to take place during a service window or using a custom schedule appropriate to specific blob usage.

- More e2e tests could also include multiple overlapping upload processes (e.g. one starts before the other has finished, two processes try to upload the same blob (shouldn't happen))

Reply:
Added to this file.